apps/patient/src/views/Main.tsx

```diff
@@ -1,86 +1,70 @@
-import { useState, useEffect, useCallback, createContext } from 'react';
+import { useState, useEffect, useCallback, createContext, useContext } from 'react';
 import { Outlet, useSearchParams, useNavigate, useLocation } from 'react-router-dom';
-import { Alert, AlertIcon, Center, CircularProgress } from '@chakra-ui/react';
+import { Center, CircularProgress } from '@chakra-ui/react';
 import { ChakraProvider } from '@chakra-ui/react';
 
 import { Order } from '../utils/models';
-import { GET_ORDER } from '../utils/queries';
-import { graphQLClient } from '../configs/graphqlClient';
+import { getOrder } from '../api/internal';
 
 import theme from '../configs/theme';
+import { setAuthHeader } from '../configs/graphqlClient';
 import { types } from '@photonhealth/sdk';
+import { AUTH_HEADER_ERRORS } from '../api/internal';
 
-const AUTH_HEADER_ERRORS = ['EMPTY_AUTHORIZATION_HEADER', 'INVALID_AUTHORIZATION_HEADER'];
-
-export const OrderContext = createContext(null);
+const OrderContext = createContext(null);
+export const useOrderContext = () => useContext(OrderContext);
 
 export const Main = () => {
   const [searchParams] = useSearchParams();
   const orderId = searchParams.get('orderId');
   const token = searchParams.get('token');
 
+  if (token) {
+    setAuthHeader(token);
```

Comment:
This is a side effect, should be wrapping this in an effect ... or would be ideally be handled via context / provider

Reply:
fixed